c/validation/adbc_validation_util.cc

```diff
@@ -141,7 +141,16 @@ int MakeSchema(struct ArrowSchema* schema, const std::vector<SchemaField>& field
   CHECK_ERRNO(ArrowSchemaSetTypeStruct(schema, fields.size()));
   size_t i = 0;
   for (const SchemaField& field : fields) {
-    CHECK_ERRNO(ArrowSchemaSetType(schema->children[i], field.type));
+    switch (field.type) {
+      case NANOARROW_TYPE_TIMESTAMP:
+        // TODO: don't hardcode unit here
+        CHECK_ERRNO(ArrowSchemaSetTypeDateTime(schema->children[i], field.type,
+                                               NANOARROW_TIME_UNIT_MICRO,
```

Comment:
`SchemaFields` are an invention of the `adbc_validation_util` header - is there a reason we use this custom struct instead of an `ArrowSchemaView`? I'm not sure if its worth adding `time_unit` to the former or porting over to the latter in the test suite

Reply:
I think I originally just wanted something that could be easily stack allocated, versus a nested heap allocated structure. I was also struggling with how to deal with validation of more complex types. Especially in the validation code where you may not always want full exact matches (maybe you don't care about the field name, etc.)

Reply:
Ah, for here...you could also just write out the code to build the schema by hand? This was meant as a convenience to cut down on boilerplate for simple schemas. But it was from a very early nanoarrow version, and nowadays I'm not sure how much boilerplate it actually saves compared to just writing it out.

Reply:
That makes sense for matching, though I think we still have a gap with that when creating test data in non-default precisions that needs to be solved

Reply:
Ah OK cool - will try that

Reply:
Yes, sorry, I realized I had gotten the context completely wrong once I clicked through and started reviewing

Reply:
Is this change still needed?